File: articles/decision-under-uncertainty-and-computational-risk/prolog/decision_risk_example.pl
expected_net_value(P, Benefit, Loss, Cost, Value) :-
    Bounded is max(0, min(1, P)),
    Value is Bounded * Benefit - Bounded * Loss - Cost.

% Example:
% ?- expected_net_value(0.42, 150, 80, 25, V).

File: articles/decision-under-uncertainty-and-computational-risk/fortran/decision_risk_example.f90
program decision_risk_example
  implicit none
  real :: p, benefit, loss, cost, value
  p = 0.42
  benefit = 150.0
  loss = 80.0
  cost = 25.0
  value = max(0.0, min(1.0, p)) * benefit - max(0.0, min(1.0, p)) * loss - cost
  print *, value
end program decision_risk_example
